configs/baremetal.h

```diff
@@ -117,8 +117,9 @@
 
 #define MBEDTLS_USE_TINYCRYPT
 #define MBEDTLS_HAVE_ASM
-#define MBEDTLS_OPTIMIZE_TINYCRYPT_ASM
-
+#if !( defined(__STRICT_ANSI__) && defined(__CC_ARM) )
+    #define MBEDTLS_OPTIMIZE_TINYCRYPT_ASM
```

Comment:
@AndrzejKurek not sure I understand : the optimization will always switched on if cc_arm defined ? Why we cannot allow to user decide if use it or not ?

Reply:
@AndrzejKurek , got the point, it is fine

Reply:
|         | STRICT   | !STRICT |
|---------|----------|---------|
| CC_ARM  | NO OPTIM | OPTIM   |
| !CC_ARM | OPTIM    | OPTIM   |